### PR TITLE
Add extra K8s arguments to select processes to instrument for Beyla component

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

- Adds some missing arguments in `discovery > services` block of `beyla.ebpf` component.
These allow to use regexps to select specific objects in a Kubernetes environment (like pods, deployments...) to instrument.

- Bumps Beyla to last version (1.6.2).

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: docs/sources/reference/components/beyla.ebpf.md [hunk 1]
```diff
@@ -120,14 +120,23 @@ It contains the following blocks:
...
 `exe_path`   | `string` | The path of the running service for Beyla automatically instrumented with eBPF. |         | no
+`k8s_namespace` | `string` | Regular expression of Kubernetes Namespaces to match.                          |         | no
+`k8s_pod_name`  | `string` | Regular expression of Kubernetes Pods to match.                           |         | no
+`k8s_deployment_name` | `string` | Regular expression of Kubernetes Deployments to match.                 |         | no
+`k8s_statefulset_name` | `string` | Regular expression of Kubernetes StatefulSets to match.                 |         | no
+`k8s_replicaset_name` | `string` | Regular expression of Kubernetes ReplicaSets to match.                 |         | no
+`k8s_daemonset_name` | `string` | Regular expression of Kubernetes DaemonSets to match.                 |         | no
+`k8s_owner_name` | `string` | Regular expression of Kubernetes owners of running Pods to match.                     |         | no
+`k8s_pod_labels` | `map(string)` | Key-value pairs of labels with keys matching Kubernetes Pods with the provided value as regular expression.             |         | no
```
Comment (rfratto) [inline docs/sources/reference/components/beyla.ebpf.md:139]: All of these arguments being prefixed with `k8s_` makes me feel like they belong in a `kubernetes` block, as blocks are the way to group related settings:

```grafana-alloy
services {
  name      = ...
  namespace = ...

  kubernetes {
    pod_name = ... 
  }
}
```

WDYT @marctc?

File: docs/sources/reference/components/beyla.ebpf.md [hunk 2]
```diff
@@ -120,14 +120,23 @@ It contains the following blocks:
...
+the instrumented processes need to match all the selector properties.
 
 Name         | Type     | Description                                                                     | Default | Required
 -------------|----------|---------------------------------------------------------------------------------|---------|---------
 `name `      | `string` | The name of the service to match.                                               |         | no
 `namespace`  | `string` | The namespace of the service to match.                                          |         | no
 `open_ports` | `string` | The port of the running service for Beyla automatically instrumented with eBPF. |         | no
 `exe_path`   | `string` | The path of the running service for Beyla automatically instrumented with eBPF. |         | no
+`k8s_namespace` | `string` | Regular expression of Kubernetes Namespaces to match.                          |         | no
+`k8s_pod_name`  | `string` | Regular expression of Kubernetes Pods to match.                           |         | no
```
Comment (rfratto) [inline docs/sources/reference/components/beyla.ebpf.md:133]: Are these mutually exclusive with setting name/namespace?

Comment (marctc) [reply docs/sources/reference/components/beyla.ebpf.md:133]: From what can I read in Beyla docs, no, it has to match everything defined in the `services` section. I added a note [here](https://github.com/grafana/alloy/pull/928/files#diff-01b6ebc9a981ba6d65085b056aa9b981225c215a8e7b7bdb8cc30e86ecc4c061R123-R124). Let me know if it's enough or needs more clarification.

File: docs/sources/reference/components/beyla.ebpf.md [hunk 3]
```diff
@@ -135,6 +138,23 @@ It is used to populate the `service.name` OTEL property and/or the `service_name
...
+-------------|----------|---------------------------------------------------------------------------------|---------|---------
+`namespace` | `string` | Regular expression of Kubernetes Namespaces to match.                          |         | no
+`pod_name`  | `string` | Regular expression of Kubernetes Pods to match.                           |         | no
+`deployment_name` | `string` | Regular expression of Kubernetes Deployments to match.                 |         | no
+`statefulset_name` | `string` | Regular expression of Kubernetes StatefulSets to match.                 |         | no
+`replicaset_name` | `string` | Regular expression of Kubernetes ReplicaSets to match.                 |         | no
+`daemonset_name` | `string` | Regular expression of Kubernetes DaemonSets to match.                 |         | no
+`owner_name` | `string` | Regular expression of Kubernetes owners of running Pods to match.                     |         | no
+`pod_labels` | `map(string)` | Key-value pairs of labels with keys matching Kubernetes Pods with the provided value as regular expression.             |         | no
```
Comment (rfratto) [inline docs/sources/reference/components/beyla.ebpf.md:155]: To align with other documentation pages these should have defaults filled out to the empty value; strings should be documented with a default of `""`, and `map(string)`s should be documented with a default of `{}`.

**Review by rfratto — APPROVED** [review]
LGTM, sorry for dropping the ball on this!

**Pull request merged**